### PR TITLE
Add TabNine cli and completion-tabnine vim plugin

### PR DESCRIPTION
**Pull request by onsails**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/tabnine/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+in stdenv.mkDerivation rec {
+  pname = "tabnine";
+  version = "3.1.1";
+
+  src = fetchurl {
+    url = "https://update.tabnine.com/${version}/${target}/TabNine";
+    sha256 = "w+Ufy4pICfQmseKCeohEQIP0VD6YrkYTEn41HX40Zlw=";
+  };
+
+  phases = [ "installPhase" ];
```
Comment (jonringer) [inline pkgs/development/tools/tabnine/default.nix:19]: specifying individual phases is deprecated
```suggestion
  dontBuild = true;
```

**Review by jonringer — COMMENTED** [review]
missing update commit

When adding new vim plugins, please follow [directions for adding new vim plugins](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md#adding-new-plugins-to-nixpkgs).

The steps should roughly be:
```
git checkout -b <branch-name>
GITHUB_API_TOKEN=[personal token] ./pkgs/misc/vim-plugins/update.py --add "[owner]/[name]"
git push <fork> <branch>
```
you should see the following in your commit history:
```
vimPlugins: update
vimPlugins.[name]: init at <date>
```
If the package needs additional patching or you want to include a meta section, please put the related changes in `pkgs/misc/vim-plugins/overrides.nix`. Do not edit the generated.nix, as those changes will be lost next update.

Why GITHUB_API_TOKEN?
Github limits github queries to 100/hr if you're a guest, but if you have a personal token, it will raise it to 5000/hr. The vim plugin update script will query the repositories to see what the latest release was, which will exceed the unauthenticated query limit.

For documentation on creating a token, please refer [to the official docs](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token#creating-a-token)

**Review by jonringer — COMMENTED** [review]
do you mind updating `vim-clap`'s cargoSha, looks like they updated the lockfile

```
https://github.com/NixOS/nixpkgs/pull/99264
2 packages marked as broken and skipped:
tabnine vimPlugins.completion-tabnine

1 package failed to build:
vimPlugins.vim-clap

41 packages built:
vimPlugins.Coqtail vimPlugins.Supertab vimPlugins.Syntastic vimPlugins.Tagbar vimPlugins.YouCompleteMe vimPlugins.ale vimPlugins.coc-explorer vimPlugins.coc-markdownlint vimPlugins.completion-nvim vimPlugins.diagnostic-nvim vimPlugins.direnv-vim vimPlugins.easymotion vimPlugins.fugitive vimPlugins.lh-brackets vimPlugins.lh-vim-lib vimPlugins.neoformat vimPlugins.neoterm vimPlugins.nvim-gdb vimPlugins.nvim-treesitter vimPlugins.polyglot vimPlugins.tmuxNavigator vimPlugins.vim-abolish vimPlugins.vim-autoformat vimPlugins.vim-cue vimPlugins.vim-elixir vimPlugins.vim-go vimPlugins.vim-gutentags vimPlugins.vim-javascript-syntax vimPlugins.vim-jsx-pretty vimPlugins.vim-lsc vimPlugins.vim-lsp vimPlugins.vim-monokai vimPlugins.vim-rails vimPlugins.vim-snippets vimPlugins.vim-test vimPlugins.vim-visual-multi vimPlugins.vim-which-key vimPlugins.vimspector vimPlugins.vimtex vimPlugins.vista-vim vimPlugins.yats-vim
```

**Comment by onsails** [comment]
> do you mind updating `vim-clap`'s cargoSha, looks like they updated the lockfile
> 
> ```
> https://github.com/NixOS/nixpkgs/pull/99264
> 2 packages marked as broken and skipped:
> tabnine vimPlugins.completion-tabnine
> 
> 1 package failed to build:
> vimPlugins.vim-clap
> 
> 41 packages built:
> vimPlugins.Coqtail vimPlugins.Supertab vimPlugins.Syntastic vimPlugins.Tagbar vimPlugins.YouCompleteMe vimPlugins.ale vimPlugins.coc-explorer vimPlugins.coc-markdownlint vimPlugins.completion-nvim vimPlugins.diagnostic-nvim vimPlugins.direnv-vim vimPlugins.easymotion vimPlugins.fugitive vimPlugins.lh-brackets vimPlugins.lh-vim-lib vimPlugins.neoformat vimPlugins.neoterm vimPlugins.nvim-gdb vimPlugins.nvim-treesitter vimPlugins.polyglot vimPlugins.tmuxNavigator vimPlugins.vim-abolish vimPlugins.vim-autoformat vimPlugins.vim-cue vimPlugins.vim-elixir vimPlugins.vim-go vimPlugins.vim-gutentags vimPlugins.vim-javascript-syntax vimPlugins.vim-jsx-pretty vimPlugins.vim-lsc vimPlugins.vim-lsp vimPlugins.vim-monokai vimPlugins.vim-rails vimPlugins.vim-snippets vimPlugins.vim-test vimPlugins.vim-visual-multi vimPlugins.vim-which-key vimPlugins.vimspector vimPlugins.vimtex vimPlugins.vista-vim vimPlugins.yats-vim
> ```

sure, will do it in a few minutes.
Where can I see this report? I don't see anything like this in github actions...

**Comment by jonringer** [comment]
I'm running `nixpkgs-review`
```
nix-shell -p nixpkgs-review --run "nixpkgs-review pr 99264"
```
it's also one of the bullets on the template

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 99264 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages marked as broken and skipped:</summary>  <ul>    <li>tabnine</li>    <li>vimPlugins.completion-tabnine</li>  </ul></details><details>  <summary>1 package failed to build:</summary>  <ul>    <li>vimPlugins.vim-clap</li>  </ul></details><details>  <summary>41 packages built:</summary>  <ul>    <li>vimPlugins.Coqtail</li>    <li>vimPlugins.Supertab (vimPlugins.supertab)</li>    <li>vimPlugins.Syntastic (vimPlugins.syntastic)</li>    <li>vimPlugins.Tagbar (vimPlugins.tagbar)</li>    <li>vimPlugins.YouCompleteMe</li>    <li>vimPlugins.ale</li>    <li>vimPlugins.coc-explorer</li>    <li>vimPlugins.coc-markdownlint</li>    <li>vimPlugins.completion-nvim</li>    <li>vimPlugins.diagnostic-nvim</li>    <li>vimPlugins.direnv-vim</li>    <li>vimPlugins.easymotion (vimPlugins.vim-easymotion)</li>    <li>vimPlugins.fugitive (vimPlugins.vim-fugitive)</li>    <li>vimPlugins.lh-brackets</li>    <li>vimPlugins.lh-vim-lib</li>    <li>vimPlugins.neoformat</li>    <li>vimPlugins.neoterm</li>    <li>vimPlugins.nvim-gdb</li>    <li>vimPlugins.nvim-treesitter</li>    <li>vimPlugins.polyglot (vimPlugins.vim-polyglot)</li>    <li>vimPlugins.tmuxNavigator (vimPlugins.tmux-navigator ,vimPlugins.vim-tmux-navigator)</li>    <li>vimPlugins.vim-abolish</li>    <li>vimPlugins.vim-autoformat</li>    <li>vimPlugins.vim-cue</li>    <li>vimPlugins.vim-elixir</li>    <li>vimPlugins.vim-go</li>    <li>vimPlugins.vim-gutentags</li>    <li>vimPlugins.vim-javascript-syntax</li>    <li>vimPlugins.vim-jsx-pretty</li>    <li>vimPlugins.vim-lsc</li>    <li>vimPlugins.vim-lsp</li>    <li>vimPlugins.vim-monokai</li>    <li>vimPlugins.vim-rails</li>    <li>vimPlugins.vim-snippets</li>    <li>vimPlugins.vim-test</li>    <li>vimPlugins.vim-visual-multi</li>    <li>vimPlugins.vim-which-key</li>    <li>vimPlugins.vimspector</li>    <li>vimPlugins.vimtex</li>    <li>vimPlugins.vista-vim</li>    <li>vimPlugins.yats-vim</li>  </ul></details>

**Comment by jonringer** [comment]
I should have built the unfree packages:
```
[nix-shell:/home/jon/.cache/nixpkgs-review/pr-99264-2]$ nix-build ./nixpkgs/ -A tabnine
these derivations will be built:
  /nix/store/jghiqgm6a1hwdws81w9dwxqdrplz84wd-TabNine.drv
  /nix/store/8va6xq5rflfvkn8vcw8vnwrb783gygnh-tabnine-3.1.1.drv
building '/nix/store/jghiqgm6a1hwdws81w9dwxqdrplz84wd-TabNine.drv'...

trying https://update.tabnine.com/3.1.1/x86_64-unknown-linux-musl/TabNine
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15.1M  100 15.1M    0     0  7414k      0  0:00:02  0:00:02 --:--:-- 7414k
hash mismatch in fixed-output derivation '/nix/store/0l29blcimkb75b87r1xqlri6sqba198p-TabNine':
  wanted: sha256:0p366iz1sdby289ldblq7rag90s08j47m0p2n4kg82a8ib5izrf3
  got:    sha256:13iram89cxym8s3q2sa925gprwsb3y70q3fxp92i9n1kcijnsac5
```

**Comment by onsails** [comment]
I'll make a new PR to fix this

**Pull request merged**